Assets/Plugins/LeapMotion/Core/Editor/LeapServiceProviderEditor.cs

```diff
@@ -61,79 +80,187 @@ public class LeapServiceProviderEditor : CustomEditorBase<LeapServiceProvider> {
+          break;
+        case LeapServiceProvider.InteractionVolumeVisualization.Automatic:
+          DetectConnectedDevice();
+          break;
+        default:
+          break;
+      }
+
+    }
+
+    private void ParseStereoIR170InteractionMeshData() {
+
+      if (_stereoIR170InteractionZoneMesh == null) {
+        _stereoIR170InteractionZoneMesh = (Mesh)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "Plugins", "LeapMotion", "Core", "Models", "StereoIR170-interaction-cone.obj"), typeof(Mesh));
+      }
+
+      if (_stereoIR170InteractionMaterial == null) {
+        _stereoIR170InteractionMaterial = (Material)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "Plugins", "LeapMotion", "Core", "Materials", "StereoIR170InteractionVolume.mat"), typeof(Material));
+      }
+
+    }
+
+    private LeapServiceProvider LeapServiceProvider {
+      get {
+
+        if (this._leapServiceProvider != null) { 
+          return this._leapServiceProvider;
+        }
+        else {
+          this._leapServiceProvider = this.target.GetComponent<LeapServiceProvider>();
+
+          return this._leapServiceProvider;
+        }
+      }
+    }
+
+    private Controller LeapController {
+      get {
+
+        if (this._leapController != null) {
+          return this._leapController;
+        }
+        else
+        {
+          this._leapController = LeapServiceProvider?.GetLeapController();
+
+          if (this._leapController != null) {
+            this._leapController.Device += _leapController_DeviceChanged;
+            this._leapController.DeviceLost += _leapController_DeviceChanged;
+          }
+
+          return this._leapController;
+        }
+      }
+    }
+
+    private void _leapController_DeviceChanged(object sender, DeviceEventArgs e) {
+      EditorWindow view = EditorWindow.GetWindow<SceneView>();
+      view.Repaint();
+    }
+
+    private void DetectConnectedDevice() {
+
+      if (LeapController?.Devices?.Count == 1)
+      {
+        if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_RIGEL) {
+          DrawStereoIR170InteractionZoneMesh();
+        }
+        else if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_PERIPHERAL) {
+          DrawLeapMotionControllerInteractionZone(LMC_BOX_WIDTH, LMC_BOX_DEPTH, LMC_BOX_RADIUS, Color.white);
+        }
+      }
+    }
+
+    private LeapServiceProvider.InteractionVolumeVisualization? GetSelectedInteractionVolume() {
+
+      return LeapServiceProvider?.SelectedInteractionVolumeVisualization;
+    }
+
+    private void DrawStereoIR170InteractionZoneMesh() {
+
+      if (_stereoIR170InteractionMaterial != null && _stereoIR170InteractionZoneMesh != null) {
+        _stereoIR170InteractionMaterial.SetPass(0);
+
+        Graphics.DrawMeshNow(_stereoIR170InteractionZoneMesh,
+           target.transform.localToWorldMatrix * Matrix4x4.TRS(_stereoIR170InteractionZoneMeshOffset, Quaternion.Euler(-90, 0, 0), Vector3.one * 0.001f));
```

Comment:
The Service Providers contain built-in offsetting systems (which modify the "device origin", affecting the interaction zone).   

Luckily, it's pretty easy to use these:
```suggestion
           target.transform.localToWorldMatrix * Matrix4x4.TRS(controllerOffset, deviceRotation * Quaternion.Euler(-90, 0, 0), Vector3.one * 0.001f));
```